src/components/common/StepMessage.jsx

```diff
@@ -156,6 +157,7 @@ class StepMessage extends Component {
       break;
 
       case 'other':
+      const segment = currentStep.segment(name, location)
```

Comment:
Ok, good, you're making progress in making changes that are re-useable, but you are breaking other pages in the process.

If you are going to make a change to a reuseable component, you should make sure all of the other components using it are also prepped for a change.

I would recommend you changing 'other' back to what it was and just add another element in the currentStep that can work as function, but leave `currentStep.segment` alone.

Reply:
```
      case 'other':
      const functionSegment = currentStep.functionSegment(name, location)
        return (
          <div>
            <Segment.Group>
              <Segment>
                <Header as='h2'>
                  {currentStep.name}
                  <Header.Subheader>
                    {currentStep.subheader}
                  </Header.Subheader>
                </Header>
              </Segment>

              {currentStep.segment}
              {currentStep.functionSegment}
            </Segment.Group>
          </div>
        );
      break;
```

Reply:
I would recommend this ^